xtraplatform-features-sql/src/main/java/de/ii/xtraplatform/feature/provider/sql/app/FilterEncoderSql.java

```diff
@@ -488,122 +477,104 @@ public String visit(Between between, List<String> children) {
-                         .replace("'0000-01-01T00:00:00Z'", "'-infinity'");
+            Instant instant = getStart(literal);
+            if (instant==Instant.MIN)
+                return sqlDialect.applyToDatetimeLiteral(sqlDialect.applyToInstantMin());
+            return sqlDialect.applyToDatetimeLiteral(instant.toString());
         }
 
         private Instant getEnd(TemporalLiteral literal) {
-            return (literal.getType() == Interval.class)
-                    ? ((Interval) literal.getValue()).getEnd()
-                    : (Instant) literal.getValue();
+            if (literal.getType() == Interval.class) {
+                Instant end = ((Interval) literal.getValue()).getEnd();
+                if (end==Instant.MAX)
+                    return end;
+                return end.minusSeconds(1);
+            } else if (literal.getType() == Instant.class) {
+                return ((Instant) literal.getValue());
+            } else {
+                return ((LocalDate) literal.getValue()).atTime(23,59,59).atZone(ZoneOffset.UTC).toInstant();
+            }
         }
 
         private String getEndAsString(TemporalLiteral literal) {
-            return String.format("TIMESTAMP '%s'", getEnd(literal))
-                         .replace("'9999-12-31T23:59:59Z'", "'infinity'");
+            Instant instant = getEnd(literal);
+            if (instant==Instant.MAX)
+                return sqlDialect.applyToDatetimeLiteral(sqlDialect.applyToInstantMax());
+            return sqlDialect.applyToDatetimeLiteral(instant.toString());
         }
 
         private Instant getEndExclusive(TemporalLiteral literal) {
-            return (literal.getType() == Interval.class)
-                    ? ((Interval) literal.getValue()).getEnd().plusSeconds(1)
-                    : (Instant) literal.getValue();
+            if (literal.getType() == Interval.class) {
+                return ((Interval) literal.getValue()).getEnd();
+            } else if (literal.getType() == Instant.class) {
+                return ((Instant) literal.getValue());
+            } else {
+                return ((LocalDate) literal.getValue()).plusDays(1).atStartOfDay(ZoneOffset.UTC).toInstant();
+            }
         }
 
         private String getEndExclusiveAsString(TemporalLiteral literal) {
-            return String.format("TIMESTAMP '%s'", getEndExclusive(literal))
-                         .replace("'+10000-01-01T00:00:00Z'", "'infinity'");
+            Instant instant = getEndExclusive(literal);
+            if (instant==Instant.MAX)
+                return sqlDialect.applyToDatetimeLiteral(sqlDialect.applyToInstantMax());
+            return sqlDialect.applyToDatetimeLiteral(instant.toString());
         }
 
         @Override
         public String visit(TemporalOperation temporalOperation, List<String> children) {
             String operator = sqlDialect.getTemporalOperator(temporalOperation.getClass());
+            if (Objects.isNull(operator))
```

Comment:
The code to handle `TIntersects` support of SQL backends is unnecessarily complex. Instead of asking the provider if it is supported and then passing that information through all application layers to build a different filter in `CqlTextVisitor`, this should be a detail handled by the encoder depending on the operator returned by the dialect being null or not.
So please remove `FeatureProvider2.supportsTIntersects` and the parameter `useTIntersects` that was added to various methods.

The other question is if it is really worth the hassle to support two different encodings. I am not sure that `OVERLAPS` is really faster. So you might remove `TIntersects` as well.

Reply:
To handle this in the encoder results in very complex code (which probably also doesn't handle all cases), which is why I went for the current approach. I will remove also TIntersects and map everything to scalar comparisons.